packages/worker/src/worker.ts

```diff
@@ -1297,27 +1304,36 @@ export class Worker {
    * This function does not throw, it will log in case of missing sinks
    * or failed sink function invocations.
    */
-  protected async processSinkCalls(externalCalls: SinkCall[], info: WorkflowInfo, isReplaying: boolean): Promise<void> {
+  protected async processSinkCalls(externalCalls: SinkCall[], isReplaying: boolean): Promise<void> {
     const { sinks } = this.options;
+
+    const filteredCalls = externalCalls
+      // Map sink call to the corresponding sink function definition
+      .map((call) => ({ call, sink: sinks?.[call.ifaceName]?.[call.fnName] }))
+      // Reject calls to undefined sink definitions
+      .filter(({ call: { ifaceName, fnName }, sink }) => {
+        if (sink !== undefined) return true;
+        this.log.error('Workflow referenced an unregistered external sink', {
+          ifaceName,
+          fnName,
+        });
+        return false;
+      })
+      // If appropriate, reject calls to sink functions not configured with `callDuringReplay = true`
+      .filter(({ sink }) => sink?.callDuringReplay || !isReplaying);
+
+    // Make a wrapper function, to make things easier afterward
     await Promise.all(
-      externalCalls.map(async ({ ifaceName, fnName, args }) => {
-        const dep = sinks?.[ifaceName]?.[fnName];
-        if (dep === undefined) {
-          this.log.error('Workflow referenced an unregistered external sink', {
-            ifaceName,
-            fnName,
+      filteredCalls.map(async ({ call, sink }) => {
+        try {
+          await sink?.fn(call.workflowInfo, ...call.args);
+        } catch (error) {
+          this.log.error('External sink function threw an error', {
+            ifaceName: call.ifaceName,
+            fnName: call.fnName,
+            error,
+            workflowInfo: call.workflowInfo,
```

Comment:
Are you passing around workflow info for each call because it can be mutated during an activation? (e.g. search attributes)